### PR TITLE
add: Day3 step 1

### PR DESCRIPTION
**Pull request by pierre-castro**

### CODE REVIEW

File: Day3/test.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+
+from day3 import step1, step2
+
+def test():
+    values = ["00100","11110","10110","10111","10101","01111","00111","11100","10000","11001","00010","01010"]
+    assert step1(values) == 198
+    print("Step 1 ok")
+
+test()
```
Comment (ade-sede) [inline Day3/test.py]:
```
import unittest
from day 3 import step1, step2

class TestDay3(unittest.TestCase):
  def test_example(self):
    values = ["00100","11110","10110","10111","10101","01111","00111","11100","10000","11001","00010","01010"]
    self.assertEqual(step1(values), 198);

if __name__ == '__main__':
    unittest.main()
```

File: Day3/test.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
```
Comment (ade-sede) [inline Day3/test.py]: Remove this line.

This isn't a script

File: Day3/day3.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
```
Comment (ade-sede) [inline Day3/day3.py:1]: Remove this line, this isn't a script

File: Day3/day3.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def step1(ll):
```
Comment (ade-sede) [inline Day3/day3.py]: As a reader who doesn't know the exercise, an anonymous name like `ll` is unclear.
It doesn't represent
- the elements inside the list
- what you plan to do with it

Use better names that convey the variable's purpose and meaning, and look into [type hints](https://docs.python.org/3/library/typing.html)

File: Day3/day3.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        sumof0 = sum(number[bit] == '0' for number in ll)
+        sumof1 = sum(number[bit] == '1' for number in ll)
+        if sumof0 > sumof1:
+            gamma += '0'
+            epsilon += '1'
+        else:
+            gamma += '1'
+            epsilon += '0'
+        
+    return int(gamma, 2) * int(epsilon, 2)
```
Comment (ade-sede) [inline Day3/day3.py:15]: Clever :tophat:

File: Day3/day3.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def step1(ll):
+    
+    gamma = ''
+    epsilon = ''
+
+    for bit in range(len(ll[0])):
+        sumof0 = sum(number[bit] == '0' for number in ll)
+        sumof1 = sum(number[bit] == '1' for number in ll)
```
Comment (ade-sede) [inline Day3/day3.py]: `sum_of_0`  
`sum_of_1`

File: Day3/day3.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        if sumof0 > sumof1:
+            gamma += '0'
+            epsilon += '1'
+        else:
+            gamma += '1'
+            epsilon += '0'
```
Comment (ade-sede) [inline Day3/day3.py]: [Ternary assignation] may be clearer here

**Review by ade-sede — CHANGES_REQUESTED** [review]

File: Day3/day3.py (hunk 3 above)
Comment (pierre-castro) [reply Day3/day3.py:1]: Got it. Need more explanations on when to use a script / python interpreter.

File: Day3/day3.py (hunk 4 above)
Comment (pierre-castro) [reply Day3/day3.py]: Changed the name `ll` to `bin_list`

File: Day3/day3.py (hunk 6 above)
Comment (pierre-castro) [reply Day3/day3.py]: Done

File: Day3/day3.py (hunk 7 above)
Comment (pierre-castro) [reply Day3/day3.py]: It is indeed much clearer.

File: Day3/day3.py (hunk 5 above)
Comment (pierre-castro) [reply Day3/day3.py:15]: Thanks 👍

File: Day3/day3.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+def step1(bin_list):    
+    gamma = ''
+    epsilon = ''
+
+    for bit in range(len(bin_list[0])):
+        sum_of_0 = sum(number[bit] == '0' for number in bin_list)
+        sum_of_1 = sum(number[bit] == '1' for number in bin_list)
+
+        gamma += '0' if (sum_of_0 > sum_of_1) else '1'
+        epsilon += '1' if (sum_of_0 > sum_of_1) else '0'
```
Comment (ade-sede) [inline Day3/day3.py]: Python style tends to avoid parentheses wherever possible
These parentheses are notrequired and can be remove

Comment (pierre-castro) [reply Day3/day3.py]: Done

**Pull request merged**